### ISSUE TITLE
One of Comments's migrations failed: Database Exception: SQLSTATE[42S01]: Base table or view already exists:

### ISSUE THREAD

**Issue by ademers** [verbb/comments#114]
### Description

I get this error when trying to install Comments for Craft 3 on a site that was originally running Craft 2.x with Comments for Craft 2.x v0.4.9 before I upgraded it to Craft 3.0.x.

Full error message: https://gist.github.com/ademers/3a1ff0c4b000758695ba2b2a90f8a8dc

### Steps to reproduce

1. Upgrade a site from Craft 2.x with Comments for Craft 2.x v0.4.9 to Craft 3.0.x.
2. Attempt to install Comments for Craft 3 from the plugin store.

### Additional info

- Plugin version: 1.0.6.3
- Craft version: 3.0.29

**Comment by engram-design** [comment]
Should be fixed in [1.0.7](https://github.com/verbb/comments/releases/tag/1.0.7), let me know if you're still having issues.

**Comment by ademers** [comment]
Hi Josh,

Just tried again with 1.0.7 and now getting following error:

```
Integrity constraint violation: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (lccphoto.#sql-47d_e8e5, CONSTRAINT craft_comments_comments_id_fk FOREIGN KEY (id) REFERENCES craft_elements (id) ON DELETE CASCADE)
The SQL being executed was: ALTER TABLE craft_comments_comments ADD CONSTRAINT craft_comments_comments_id_fk FOREIGN KEY (id) REFERENCES craft_elements (id) ON DELETE CASCADE

Migration: verbb\comments\migrations\m180825_000000_craft3_upgrade

Output:

> add foreign key craft_comments_comments_id_fk: {{%comments_comments}} (id) references {{%elements}} (id) ...Exception: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (lccphoto.#sql-47d_e8e5, CONSTRAINT craft_comments_comments_id_fk FOREIGN KEY (id) REFERENCES craft_elements (id) ON DELETE CASCADE)
The SQL being executed was: ALTER TABLE craft_comments_comments ADD CONSTRAINT craft_comments_comments_id_fk FOREIGN KEY (id) REFERENCES craft_elements (id) ON DELETE CASCADE (/home/vagrant/sites/lccphoto/vendor/yiisoft/yii2/db/Schema.php:664)
#0 /home/vagrant/sites/lccphoto/vendor/yiisoft/yii2/db/Command.php(1263): yii\db\Schema->convertException(Object(PDOException), 'ALTER TABLE cr...')
#1 /home/vagrant/sites/lccphoto/vendor/yiisoft/yii2/db/Command.php(1075): yii\db\Command->internalExecute('ALTER TABLE cr...')
#2 /home/vagrant/sites/lccphoto/vendor/yiisoft/yii2/db/Migration.php(467): yii\db\Command->execute()
#3 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/db/Migration.php(382): yii\db\Migration->addForeignKey('craft_comments_...', '{{%comments_com...', 'id', '{{%elements}}', 'id', 'CASCADE', NULL)
#4 /home/vagrant/sites/lccphoto/vendor/verbb/comments/src/migrations/m180825_000000_craft3_upgrade.php(37): craft\db\Migration->addForeignKey('craft_comments_...', '{{%comments_com...', 'id', '{{%elements}}', 'id', 'CASCADE', NULL)
#5 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/db/Migration.php(56): verbb\comments\migrations\m180825_000000_craft3_upgrade->safeUp()
#6 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/db/MigrationManager.php(243): craft\db\Migration->up(true)
#7 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/db/MigrationManager.php(163): craft\db\MigrationManager->migrateUp(Object(verbb\comments\migrations\m180825_000000_craft3_upgrade))
#8 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/services/Updates.php(215): craft\db\MigrationManager->up()
#9 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/controllers/BaseUpdaterController.php(443): craft\services\Updates->runMigrations(Array)
#10 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/controllers/pluginstore/InstallController.php(158): craft\controllers\BaseUpdaterController->runMigrations(Array)
#11 [internal function]: craft\controllers\pluginstore\InstallController->actionMigrate()
#12 /home/vagrant/sites/lccphoto/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#13 /home/vagrant/sites/lccphoto/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#14 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/web/Controller.php(104): yii\base\Controller->runAction('migrate', Array)
#15 /home/vagrant/sites/lccphoto/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('migrate', Array)
#16 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/web/Application.php(282): yii\base\Module->runAction('pluginstore/ins...', Array)
#17 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/web/Application.php(664): craft\web\Application->runAction('pluginstore/ins...')
#18 /home/vagrant/sites/lccphoto/vendor/craftcms/cms/src/web/Application.php(236): craft\web\Application->_processUpdateLogic(Object(craft\web\Request))
#19 /home/vagrant/sites/lccphoto/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#20 /home/vagrant/sites/lccphoto/web/index.php(21): yii\base\Application->run()
#21 {main}
```

**Comment by engram-design** [comment]
Hmmm, looks like some comments element's have been deleted somewhere along the line? I'm not sure if your install had some foreign key issues on Craft 2?

Hopefully [1.0.7.1](https://github.com/verbb/comments/releases/tag/1.0.7.1) should get you past the migration.

**Comment by ademers** [comment]
That did it! Thanks!

**Comment by ademers** [comment]
I was finally able to install Comments for Craft 3 in local dev (Homestead). However, when I view the comments, I see `[Deleted element]` https://cl.ly/6f927d5e24b2

When I view comment details, the Element's URL points to the Production server. https://cl.ly/39556a550b11

What am I missing?

**Comment by engram-design** [comment]
Is that for all comments, or just a selection? That means that Comments can't find the element that the comment was made on (entry or otherwise). Might seem like a stupid question, but those element's still exist right? If so, something might be going on, and its worth getting in touch via the Slack Discord to see what the issue is...

**Comment by engram-design** [comment]
@ademers Should be fixed in [1.1.0](https://github.com/verbb/comments/releases/tag/1.1.0)
